### DISCUSSION TITLE
Blitz Recipes

### DISCUSSION THREAD

**Discussion by aem** [blitz-js/blitz#797] [Ideas]
Blitz Recipes are almost production-ready! We've got a working recipe to install Tailwind CSS, and @Zeko369 is working on one for Chakra. What other libraries would you like to see an automated installer for? This list will be updated based on the discussion below

- [x] Tailwind CSS
- [x] Chakra @Zeko369 
- [x] Theme UI @aem 
- [ ] Cypress @aem 
- [x] Emotion
- [x] Styled-components
- [x] Render.com
- [ ] Google analytics
- [ ] Fathom analytics
- [ ] Sentry
- [ ] MDX
- [x] Material UI

**Comment by flybayer** [comment]
Emotion
Styled-components
Render.com (the Render.yaml for deployment)
Google analytics
Fathom analytics 
Sentry
MDX

**Reply by flybayer** [reply]
Actually wait on Sentry until I finalize the error boundary. 

Add material UI

**Comment by Zeko369** [comment]
I'll also try adding PM2 for deployment in a couple of days and maybe play around with dokku.

**Comment by LoriKarikari** [comment]
I'll try adding a multi-stage Docker setup if that's fine?

**Reply by flybayer** [reply]
Yes! Already a couple options here: https://github.com/blitz-js/blitz/issues/289

**Reply by LoriKarikari** [reply]
These examples are pretty basic and missing some stuff if you want a multi-stage setup. Should I create a full fledged setup and users should delete all the unwanted stuff themselves? I can imagine not everyone needs an advanced setup. That's why I was asking in Slack if it was possible to add prompts and stuff. So you could customise the setup.

**Reply by flybayer** [reply]
Good question. Probably if someone is using docker, they will know enough to half-way understand the file and remove things if need.

So I think add everything, and comment it well.

**Comment by DevanB** [comment]
- Stripe
- i18n (I think Next struggles with this still, so might be hard)
- A worker queue solution

These would all be awesome recipes and Blitz on the map even more for public consumption.

**Reply by Zeko369** [reply]
I think that next is working on build in / first party solution for i18n so it might be smart to wait for that or just add next-i18n

**Comment by wKovacs64** [comment]
I'm trying Emotion but I don't know anything about this AST stuff yet, so it's slow going. 🙂

**Reply by flybayer** [reply]
This may help if you haven't found it already: https://astexplorer.net/

**Reply by wKovacs64** [reply]
Oh yeah, leaning on that heavily already. 😁

**Comment by remidej** [comment]
I've never used [Storybook](https://storybook.js.org/), but I think it can be a good match for Recipes

**Reply by enricoschaaf** [reply]
Maybe, a Preact recipe, using it for quite some time and has never let me down.

**Comment by LoriKarikari** [comment]
Maybe we can add a recipe too that creates a database on Digital Ocean? This can be done with Pulumi which is an IAC tool in TypeScript (I think). I can take a look at that when I have some time. Still need to finish the Docker recipe too... 😅

**Reply by flybayer** [reply]
Ooo, that could be cool

**Reply by enricoschaaf** [reply]
Maybe we could to something similar with Heroku, maybe on blitz new you choose your database provider and your hosting provider and when you deploy for the first time the build tool uses the Heroku CLI to create a free database, swap the url, run the migration.🤔

**Comment by MrLeebo** [comment]
What about a recipe for websockets, like Pusher or Socket.io?

**Comment by falleng0d** [comment]
Ant Design like materialui is a huge ui framework. It has some considerations when setting it up with nextjs. I belive having a template for it will be very useful. I can even provide a configuration of my own

**Comment by openspaceranger** [comment]
Are all the recipes still usable post pivot? The reason I ask is because the material-ui recipe has not been updated for two years, and I was getting some strange errors:

```
 ⚠ ./src/app/blitz-server.ts
Attempted import error: 'setupBlitzServer' is not exported from '@blitzjs/next' (imported as 'setupBlitzServer').

Import trace for requested module:
./src/app/blitz-server.ts
./src/app/page.tsx

./src/app/blitz-server.ts
Attempted import error: 'AuthServerPlugin' is not exported from '@blitzjs/auth' (imported as 'AuthServerPlugin').

Import trace for requested module:
./src/app/blitz-server.ts
./src/app/page.tsx

./src/app/blitz-server.ts
Attempted import error: 'PrismaStorage' is not exported from '@blitzjs/auth' (imported as 'PrismaStorage').

Import trace for requested module:
./src/app/blitz-server.ts
./src/app/page.tsx

./src/app/blitz-server.ts
Attempted import error: 'simpleRolesIsAuthorized' is not exported from '@blitzjs/auth' (imported as 'simpleRolesIsAuthorized').

Import trace for requested module:
./src/app/blitz-server.ts
./src/app/page.tsx

./src/app/blitz-server.ts
Attempted import error: 'RpcServerPlugin' is not exported from '@blitzjs/rpc' (imported as 'RpcServerPlugin').

Import trace for requested module:
./src/app/blitz-server.ts
./src/app/page.tsx

./src/app/blitz-server.ts
Attempted import error: 'BlitzLogger' is not exported from 'blitz' (imported as 'BlitzLogger').

Import trace for requested module:
./src/app/blitz-server.ts
./src/app/page.tsx
2024-01-16 05:55:24.931 INFO    [blitz-invoke]  getCurrentUser() Starting with input: null
2024-01-16 05:55:24.939 DEBUG   [blitz-invoke]  getCurrentUser() Result: null
2024-01-16 05:55:24.943 INFO    [blitz-invoke]  getCurrentUser() Finished: resolver:12ms serializer:0ms total:12ms
 ⚠ ./src/app/blitz-server.ts
Attempted import error: 'setupBlitzServer' is not exported from '@blitzjs/next' (imported as 'setupBlitzServer').

Import trace for requested module:
./src/app/blitz-server.ts
./src/app/page.tsx

./src/app/blitz-server.ts
Attempted import error: 'AuthServerPlugin' is not exported from '@blitzjs/auth' (imported as 'AuthServerPlugin').

Import trace for requested module:
./src/app/blitz-server.ts
./src/app/page.tsx

./src/app/blitz-server.ts
Attempted import error: 'PrismaStorage' is not exported from '@blitzjs/auth' (imported as 'PrismaStorage').

Import trace for requested module:
./src/app/blitz-server.ts
./src/app/page.tsx

./src/app/blitz-server.ts
Attempted import error: 'simpleRolesIsAuthorized' is not exported from '@blitzjs/auth' (imported as 'simpleRolesIsAuthorized').

Import trace for requested module:
./src/app/blitz-server.ts
./src/app/page.tsx

./src/app/blitz-server.ts
Attempted import error: 'RpcServerPlugin' is not exported from '@blitzjs/rpc' (imported as 'RpcServerPlugin').

Import trace for requested module:
./src/app/blitz-server.ts
./src/app/page.tsx

./src/app/blitz-server.ts
Attempted import error: 'BlitzLogger' is not exported from 'blitz' (imported as 'BlitzLogger').
```
